erts/emulator/beam/erl_msacc.c

```diff
@@ -364,20 +364,18 @@ erts_msacc_request(Process *c_p, int action, Eterm *threads)
     msaccrp->ref = STORE_NC(&hp, NULL, ref);
     msaccrp->req_sched = esdp->no;
 
-    *threads = erts_no_schedulers;
-    *threads += 1; /* aux thread */
+    *threads = erts_no_aux_work_threads;
 
     erts_atomic32_init_nob(&msaccrp->refc,(erts_aint32_t)*threads);
 
     erts_proc_add_refc(c_p, *threads);
 
     if (erts_no_schedulers > 1)
```

Comment:
This condition is incorrect. I think it should be removed all together?

As it is now the emulator -> statistics_SUITE -> msacc testcase breaks when running using `+S 1`.

Reply:
Thanks! Fixed now.

Reply:
> Seems to me like the entire if should be removed?

Yes you are right, the whole if should be removed.

> Won't the fix you made break the case where we have only 1 aux thread and more than 1 schedulers?

No `erts_no_aux_work_threads` is always greater than 1 (threads handling aux work including schedulers) so it is an unnecessary test.